pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -203,12 +211,27 @@ func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 	}
 
 	processedVolumesForFSResize := sets.NewString()
+	skippedDueToPVCsUnderDeletion := false
 	for _, pod := range dswp.podManager.GetPods() {
 		if dswp.isPodTerminated(pod) {
 			// Do not (re)add volumes for terminated pods
 			continue
 		}
-		dswp.processPodVolumes(pod, mountedVolumesForPod, processedVolumesForFSResize)
+		if dswp.processPodVolumes(pod, mountedVolumesForPod, processedVolumesForFSResize) {
+			skippedDueToPVCsUnderDeletion = true
+		}
+	}
+	return skippedDueToPVCsUnderDeletion
+}
+
+func (dswp *desiredStateOfWorldPopulator) cleanupPVCsUnderDeletion(pod *v1.Pod) {
+	// Process volume spec for each volume defined in pod
+	if pod != nil && len(pod.Spec.Volumes) > 0 {
+		for _, podVolume := range pod.Spec.Volumes {
+			if pvcSource := podVolume.VolumeSource.PersistentVolumeClaim; pvcSource != nil {
+				delete(dswp.pvcsUnderDeletion, pod.Namespace+"/"+pvcSource.ClaimName)
```

Comment:
Multiple pods can share the same PVC. This structure may need to be per pod.

Reply:
I thought about this.

If an entry is taken out from pvcsUnderDeletion, the worst case is that there is one more GET to api-server after which the entry may be added.
It seems that is fine, given the current data structure.